sdk/cognitivelanguage/azure-ai-language-questionanswering/tests/test_update_knowledge_sources.py

```diff
@@ -0,0 +1,124 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import pytest
+
+from azure.core.exceptions import HttpResponseError, ClientAuthenticationError
+from azure.core.credentials import AzureKeyCredential
+
+from testcase import (
+    QuestionAnsweringTest,
+    GlobalQuestionAnsweringAccountPreparer,
+    QnaAuthoringHelper
+)
+
+from azure.ai.language.questionanswering.projects import QuestionAnsweringProjectsClient
+
+class SourcesQnasSynonymsTests(QuestionAnsweringTest):
+
+    @GlobalQuestionAnsweringAccountPreparer()
+    def test_add_source(self, qna_account, qna_key):
```

Comment:
Once we're happy that these are good to go, we could duplicate them in async?

Reply:
should also create samples/tests for async client?

Reply:
Yup!